### PR TITLE
debug_value instructions should not block analysis of static initializers

### PR DESCRIPTION
**Pull request by bob-wilson**
#### What's in this pull request?

The analyzeStaticInitializer function has a list of specific SIL instructions that it expects to see in static initializers. Anything unexpected blocks the analysis and inhibits optimizations, such as propagating global let variables. Add debug_value instructions to that list so that they do not inhibit optimizations.
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

rdar://problem/26411312

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by bob-wilson** [comment]
@swift-ci Please test and merge

File: lib/SIL/SILGlobalVariable.cpp [hunk 1]
```diff
@@ -129,6 +129,7 @@ static bool analyzeStaticInitializer(SILFunction *F, SILInstruction *&Val,
       if (I.getKind() != ValueKind::ReturnInst &&
```
Comment (slavapestov) [inline lib/SIL/SILGlobalVariable.cpp]: What about EnumInst?

Comment (swiftix) [inline lib/SIL/SILGlobalVariable.cpp]: @bob-wilson Slava makes a good point. It should be pretty easy to add support for EnumInst where payloads are e.g. simple integers.

**Comment by bob-wilson** [comment]
The Linux failures here don't seem to have anything to do with my change. I'm going to go ahead and merge this.

Thanks for the suggestion, Slava. I'll look into adding support for EnumInst as well.

**Pull request merged**